File: Android/src/main/java/com/thinkincode/utils/display/ColorUtils.java
package com.thinkincode.utils.display;

/**
 * Provides helper methods for converting between different colour representations.
 */
public class ColorUtils {

    /**
     * Converts the provided color to its #RRGGBB Hex String representation.
     * <p>
     * The alpha component of the color is ignored.
     *
     * @param color the color to convert.
     * @return the #RRGGBB Hex String representation of the provided color.
     */
    public static String convertColorToRGBHexString(int color) {
        return String.format("#%06X", (0xFFFFFF & color));
    }

    /**
     * Converts the provided color to its #AARRGGBB Hex String representation.
     *
     * @param color the color to convert.
     * @return the #AARRGGBB Hex String representation of the provided color.
     */
    public static String convertColorToARGBHexString(int color) {
        return String.format("#%08X", color);
    }
}

File: Android/src/main/java/com/thinkincode/utils/display/ColorUtils.kt
package com.thinkincode.utils.display

/**
 * Converts the color to its #RRGGBB Hexadecimal String representation.
 *
 * The alpha component of the color is ignored.
 *
 * @return the #RRGGBB Hexadecimal String representation of the color.
 */
fun Int.toRGBHexadecimalString(): String {
    return String.format("#%06X", 0xFFFFFF and this)
}

/**
 * Converts the color to its #AARRGGBB Hexadecimal String representation.
 *
 * @return the #AARRGGBB Hexadecimal String representation of the color.
 */
fun Int.toARGBHexadecimalString(): String {
    return String.format("#%08X", this)
}
